Automattic/jetpack
# Makes Sitemaps module use DOM library.

Fixes #7265 

Still to do:
- [x] If this module is enabled and there is no XML support, issue a notice instead of fatalling.
- [x] make sure it's backwards compatible with old sitemaps that were stored on dotorg sites before.
- [x] use some other algorithm of calculating sitemaps length instead of generating the whole document every time.

#### Changes proposed in this Pull Request:
* Adds unit test suite to make sure we run Sitemaps unit tests on every Travis run.
* Fixes these unit tests.
* Enables using DOM library to generate XML files instead of manual concatenation.
* Removes all early escaping to avoid overescaping bugs.
* Stores sitemaps data in custom posts using base64 encoding.
* Requiring users to have XML module enabled in PHP.

#### Testing instructions:
* Check out this PR.
* Make some test posts with special characters in titles.
* Upload some images with special characters in titles and captions.
* Upload some videos (non-VaultPress so far) with special characters in titles and captions.
* Re-generate your sitemap, you can do that by disabling and enabling the Sitemaps module.
* Make sure every generated sitemap works as expected: no YSOD, shows fine in the browser.


#### Proposed changelog entry for your changes:
* Optimized Sitemaps generation to speed them up and make them more reliable.
